ui/components/K8sTerminal/PodSelector.js

```diff
@@ -0,0 +1,286 @@
+      const response = await fetch(
+        `/api/system/kubernetes/namespaces?context=${contextId}`,
+      );
+      if (!response.ok) throw new Error('Failed to fetch namespaces');
+      const data = await response.json();
+      setNamespaces(data.namespaces || []);
+    } catch (err) {
+      setError(err.message);
+      console.error('Error fetching namespaces:', err);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const fetchPods = async (namespace) => {
+    setLoading(true);
+    setError(null);
+    try {
+      // TODO: Replace with actual GraphQL query to fetch pods
+      const response = await fetch(
+        `/api/system/kubernetes/pods?context=${contextId}&namespace=${namespace}`,
+      );
+      if (!response.ok) throw new Error('Failed to fetch pods');
+      const data = await response.json();
+      setPods(data.pods || []);
+    } catch (err) {
+      setError(err.message);
+      console.error('Error fetching pods:', err);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const extractContainers = (pod) => {
+    // Extract containers from pod spec
+    const containerList = [];
+    if (pod.spec && pod.spec.containers) {
+      containerList.push(...pod.spec.containers.map((c) => ({ name: c.name, type: 'container' })));
+    }
+    if (pod.spec && pod.spec.initContainers) {
+      containerList.push(
+        ...pod.spec.initContainers.map((c) => ({ name: c.name, type: 'initContainer' })),
+      );
+    }
+    setContainers(containerList);
+    // Auto-select first container if only one
+    if (containerList.length === 1) {
+      setSelectedContainer(containerList[0]);
+    }
+  };
+
+  const handleOpenTerminal = () => {
+    if (!selectedNamespace || !selectedPod) {
+      setError('Please select namespace and pod');
+      return;
+    }
+    setShowTerminal(true);
+  };
+
+  const handleOpenLogs = () => {
+    if (!selectedNamespace || !selectedPod) {
+      setError('Please select namespace and pod');
+      return;
+    }
+    setShowLogs(true);
+  };
+
+  const handleClose = () => {
+    setShowTerminal(false);
+    setShowLogs(false);
+    setSelectedNamespace(null);
+    setSelectedPod(null);
+    setSelectedContainer(null);
+    setContainers([]);
+    setPods([]);
+    setError(null);
+    onClose();
+  };
+
+  const handleTabChange = (event, newValue) => {
+    setTabValue(newValue);
+  };
+
+  if (showTerminal) {
+    return (
+      <Dialog open={true} onClose={() => setShowTerminal(false)} maxWidth="xl" fullWidth>
+        <K8sTerminal
+          namespace={selectedNamespace.name}
+          podName={selectedPod.metadata.name}
+          containerName={selectedContainer?.name || ''}
+          contextId={contextId}
+          onClose={() => setShowTerminal(false)}
+        />
+      </Dialog>
+    );
+  }
+
+  if (showLogs) {
+    return (
+      <Dialog open={true} onClose={() => setShowLogs(false)} maxWidth="xl" fullWidth>
+        <K8sLogsViewer
```

Comment:
Don't include "K8s" in any of the variable/component names. Understand that Meshery manages many platforms.